BootloaderCommonPkg/Library/Ext23Lib/Ext2FsLs.c

```diff
@@ -0,0 +1,239 @@
+  2. Redistributions in binary form must reproduce the above copyright
+     notice, this list of conditions and the following disclaimer in the
+     documentation and/or other materials provided with the distribution.
+  3. Neither the Name of the University nor the names of its contributors
+     may be used to endorse or promote products derived from this software
+     without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND
+  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+  ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
+  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+  OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+  HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+  OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+  SUCH DAMAGE.
+
+
+  Copyright (c) 1990, 1991 Carnegie Mellon University
+  All Rights Reserved.
+
+  Author: David Golub
+
+  Permission to use, copy, modify and distribute this software and its
+  documentation is hereby granted, provided that both the copyright
+  notice and this permission notice appear in all copies of the
+  software, derivative works or modified versions, and any portions
+  thereof, and that both notices appear in supporting documentation.
+
+  CARNEGIE MELLON ALLOWS FREE USE OF THIS SOFTWARE IN ITS "AS IS"
+  CONDITION.  CARNEGIE MELLON DISCLAIMS ANY LIABILITY OF ANY KIND FOR
+  ANY DAMAGES WHATSOEVER RESULTING FROM THE USE OF THIS SOFTWARE.
+
+  Carnegie Mellon requests users of this software to return to
+
+   Software Distribution Coordinator  or  Software.Distribution@CS.CMU.EDU
+   School of Computer Science
+   Carnegie Mellon University
+   Pittsburgh PA 15213-3890
+
+  any improvements or extensions that they make and grant Carnegie the
+  rights to redistribute these changes.
+
+    Stand-alone FILE reading package for Ext2 FILE system.
+**/
+
+#include <Library/DebugLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/MediaAccessLib.h>
+#include "Ext2Fs.h"
+#include "LibsaFsStand.h"
+
+#define NELEM(x) (sizeof (x) / sizeof(*x))
+
+typedef struct ENTRY ENTRY;
+struct ENTRY {
+  ENTRY    *EntryNext;
+  INODE32  EntryInode;
+  UINT8    EntryType;
+  CHAR8    EntryName[EXT2FS_MAXNAMLEN];
+};
+
+STATIC CONST CHAR8 *CONST mTypeStr[] = { "unknown", "REG",  "DIR",  "CHR",  "BLK",  "FIFO",  "SOCK",  "LNK" };
+
+/**
+  List directories or files and print them
+
+  @param File       pointer to an file private data
+  @param Pattern    not used for now
+
+**/
+EFI_STATUS
+EFIAPI
+Ext2fsLs (
```

Comment:
Add  IN / OUT  for parameters

Reply:
Sure, let me add it.

Reply:
I have updated all Ext23Lib files in 2nd patch.